AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet3Solutions.lean

```diff
@@ -0,0 +1,144 @@
+/-
+This is a solutions sheet.
+-/
+
+import Mathlib.Tactic
+
+/-Credit for some exercises goes to Kevin Buzzard and his Formalizing Mathematics
+course OR to the Natural Number Game. If you would like to learn more about Lean,
+Buzzard's course goes more in depth in relation to numerous undergraduate math topics.
+When exercises are from either of these sources, they will be clearly marked so as
+to give credit.-/
+
+/-In the Natural Number Game, you used the tactics:
+
+left
+right
+
+Let's do a couple warm up exercises from Formalizing Mathematics using these.
+Note that the ∨ symbol means "or," just as it did in NNG.-/
+
+variable (P Q R S T : Prop)
+
+/-FM Section 1 Sheet 6-/
+example : P → P ∨ Q := by
+  intro h
+  left
+  exact h
+  done
+
+/-FM Section 1 Sheet 6-/
+example : Q → P ∨ Q := by
+  intro h
+  right
+  exact h
+  done
+
+/-Also in the Natural Number Game, you learned a tactic called cases. However,
+the functionality and syntax you learned is actually closer to the tactic cases'
+(called cases prime), which is a more specific functionality of the broader tactic.
+Let's now go over how to use the cases tactic.
+
+First, let's look at an example:-/
+
+/-FM Section 1 Sheet 4-/
+example : P ∧ Q → P := by
+  intro h1
+  cases h1 with
+  | intro left right =>
+    exact left
+  done
+
+/-Note that the structure of the tactic looks a bit different. A shortcut to
+getting this structure automatically generated rather than having to type it all
+out is to type "cases h1 (or whatever you want to do cases on) with" and wait for
```

Comment:
This sentence can be made more precise. "whatever" can be replaced with "hypothesis", for example.